runtime/kubernetes/pod_tracker.go

```diff
@@ -143,6 +170,83 @@ func (p *podTracker) getTrackedPod(obj interface{}) *v1.Pod {
 	return pod
 }
 
+// HandleEventAdd is an AddFunc for cache.ResourceEventHandlerFuncs for Events.
+func (p *podTracker) HandleEventAdd(newObj interface{}) {
+	newEvent := p.getTrackedPodEvent(newObj)
+	if newEvent == nil {
+		// not valid or not for our tracked pod
+		return
+	}
+
+	p.Logger.Tracef(
+		"handling %s event add event for %s: fieldPath=%v",
+		newEvent.Type, // Normal, Warning
+		p.TrackedPod,
+		newEvent.InvolvedObject.FieldPath,
+	)
+
+	p.inspectContainerEvent(newEvent)
+}
+
+// HandleEventUpdate is an UpdateFunc for cache.ResourceEventHandlerFuncs for Events.
+func (p *podTracker) HandleEventUpdate(oldObj, newObj interface{}) {
+	oldEvent := p.getTrackedPodEvent(oldObj)
+	newEvent := p.getTrackedPodEvent(newObj)
+
+	if oldEvent == nil || newEvent == nil {
+		// not valid or not for our tracked pod
+		return
+	}
+
+	p.Logger.Tracef(
+		"handling %s event update event for %s: fieldPath=%v",
+		newEvent.Type, // Normal, Warning
+		p.TrackedPod,
+		newEvent.InvolvedObject.FieldPath,
+	)
+
+	p.inspectContainerEvent(newEvent)
+}
+
+// HandleEventDelete is an DeleteFunc for cache.ResourceEventHandlerFuncs for Events.
```

Comment:
If this is dead code, I'd prefer we delete it.

Reply:
deleted